tests/test_lzip_decoder.c

```diff
@@ -0,0 +1,454 @@
+///////////////////////////////////////////////////////////////////////////////
+//
+/// \file       test_lzip_decoder.c
+/// \brief      Tests decoding lzip data
+//
+//  Author:     Jia Tan
+//
+//  This file has been put into the public domain.
+//  You can do whatever you want with this file.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include "tests.h"
+
+#ifdef HAVE_LZIP_DECODER
+
+// Memlimit large enough to pass all of the test files
+#define MEMLIMIT (1U << 20)
+#define DECODE_CHUNK_SIZE 1024
+
+
+// Avoiding using the ASCII string because of potential EBCDIC system
+// compatibility. The string is: "Hello\nWorld!\n"
+static const char hello_world[] = {'H', 'e', 'l', 'l', 'o', '\n',
+			'W', 'o', 'r', 'l', 'd', '!', '\n', 0};
+// The string is "Trailing garbage\n"
+static const char trailing_garbage[] = {'T', 'r', 'a', 'i', 'l', 'i',
+		'n', 'g', ' ', 'g', 'a', 'r', 'b', 'a', 'g', 'e', '\n', 0};
+
```

Comment:
Putting ASCII chars as separate chars instead of as a string doesn't change anything. On EBCDIC systems the source files are mass-converted from ASCII to EBCDIC before compilation and thus the characters map to different byte values. I think it's enough to check for trailing garbage string and that can be done by comparing CRC32 value, that is, compare the length of the garbage and use lzma_crc32() and compare against known hardcoded value (no need for a #define for the hardcoded value). It's ugly but then it's portable and there's no need for a ASCII string in hex representation.

Reply:
I suppose the simple tedious solution is to convert the ASCII characters to hex and hardcode those, but I like hardcoding the CRC better since we shouldn't ever regenerate the test file with a new checksum type. I will work on implementing that